cmd/kritis/admission/main_test.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"crypto/tls"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/grafeas/kritis/pkg/kritis/admission"
+	"github.com/grafeas/kritis/pkg/kritis/testutil"
+)
+
+const (
+	RandomPortAddress = ":0"
+	// TODO: Generate these certs via crypt/tls lib.
+	CertFile = "./certs/cert.pem"
+	KeyFile  = "./certs/key.pem"
+)
+
+var tcIntTest = []struct {
+	name             string
+	certFile         string
+	keyFile          string
+	expectedHttpResp int
+	shouldError      bool
+}{
+	{"secure connection", CertFile, KeyFile, http.StatusOK, false},
+	{"insecure connection", "", "", http.StatusContinue, true},
+}
+
+func TestHTTPSServer(t *testing.T) {
+	http.HandleFunc("/", admission.AdmissionReviewHandler)
+	for _, tc := range tcIntTest {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
These tests are probably a bit overkill - we're just making a single call to the golang https server implementation.

Reply:
Yes that is true. I just wanted a smoke test. I can remove this.